src/go/rpk/pkg/tuners/ballast/tuner.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// +build linux
+
+package ballast
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/docker/go-units"
+	"github.com/pkg/errors"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/tuners"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/tuners/executors"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/tuners/executors/commands"
+	"golang.org/x/sys/unix"
+)
+
+type ballastTuner struct {
+	conf     config.Config
+	executor executors.Executor
+}
+
+func NewBallastFileTuner(
+	conf config.Config, executor executors.Executor,
+) tuners.Tunable {
+	return &ballastTuner{conf, executor}
+}
+
+func (t *ballastTuner) Tune() tuners.TuneResult {
+	path := config.DefaultBallastFilePath
+	if t.conf.Rpk.BallastFilePath != "" {
+		path = t.conf.Rpk.BallastFilePath
+	}
+	abspath, err := filepath.Abs(path)
+	if err != nil {
+		return tuners.NewTuneError(fmt.Errorf(
+			"couldn't resolve the absolute file path for %s: %w",
+			path,
+			err,
+		))
+	}
+
+	size := config.DefaultBallastFileSize
+	if t.conf.Rpk.BallastFileSize != "" {
+		size = t.conf.Rpk.BallastFileSize
+	}
+	sizeBytes, err := units.FromHumanSize(size)
+	if err != nil {
+		return tuners.NewTuneError(fmt.Errorf(
+			"'%s' is not a valid size unit.",
+			size,
+		))
+	}
+
+	cmd := commands.NewWriteSizedFileCmd(abspath, sizeBytes)
+	err = t.executor.Execute(cmd)
+	if err != nil {
+		return tuners.NewTuneError(err)
+	}
+	return tuners.NewTuneResult(false)
+}
+
+func (*ballastTuner) CheckIfSupported() (supported bool, reason string) {
+	return true, ""
+}
+
+func executeBallast(path string, size int64) error {
+	_, err := os.Stat(path)
+	if err == nil {
+		// If the file exists, error out to prevent overwriting important files
+		// or filesystems.
+		return fmt.Errorf("file '%s' already exists. If you're sure you want"+
+			" to replace it, remove it first and then run this command again.",
+			path,
+		)
+	}
+	if !os.IsNotExist(err) {
+		return err
+	}
+
+	f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE, 0666)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	if err := unix.Fallocate(int(f.Fd()), 0, 0, size); err != nil {
+		return fmt.Errorf("couldn't allocate the requested size while"+
+			" creating the ballast file at '%s': %w",
+			path,
+			err,
+		)
+	}
+	return errors.Wrap(f.Sync(), "couldn't sync the ballast file at "+path)
```

Comment:
Since the advent of `%w` in fmt, the errors package is less necessary. What do you think of
```go
if err := f.Sync(); err != nil {
    return fmt.Errorf("unable to sync the ballast file at %s: %w", path, err)
}
return nil
```

Reply:
Yeah, this was the 1st pass, but the one-liner looked ok to me. If you think it hinders clarity, I can revert it.

Reply:
Done.

Reply:
Cool awesome! I think moreso that errors was used pre go1.13 and I don't see it used much anymore. Although it could still be useful, I lean heavily to %w with fmt :)